src/main/java/utils/parser/OptionsBuilder.java

```diff
@@ -0,0 +1,197 @@
+    public static final String DECK_ACTION = "deck";
+    public static final String EDIT_ACTION = "edit";
+    public static final String LIST_ACTION = "list";
+
+    private static String model = null;
+    private static String action = null;
+
+    public OptionsBuilder(String model, String action) {
+        this.model = model;
+        this.action = action;
+    }
+
+    public Options buildOptions() {
+        switch (model) {
+            case CARD_MODEl:
+                switch (action) {
+                    case ADD_ACTION:
+                        return buildAddOptions();
+                    case TAG_ACTION:
+                        return buildTagOptions();
+                    case VIEW_ACTION:
+                        return buildViewOptions();
+                    case DELETE_ACTION:
+                        return buildDeleteOptions(CARD_MODEl);
+                    case DECK_ACTION:
+                        return buildDeckOptions(CARD_MODEl);
+                }
+            case TAG_MODEL:
+                switch (action) {
+                    case DELETE_ACTION:
+                        return buildDeleteOptions(TAG_MODEL);
+                    case DECK_ACTION:
+                        return buildDeckOptions(TAG_MODEL);
+                    case EDIT_ACTION:
+                        return buildEditOptions(TAG_MODEL);
+                    case LIST_ACTION:
+                        return buildListOptions(TAG_MODEL);
+                }
+            case DECK_MODEL:
+                switch (action) {
+                    case DELETE_ACTION:
+                        return buildDeleteOptions(DECK_MODEL);
+                    case EDIT_ACTION:
+                        return buildEditOptions(DECK_MODEL);
+                    case LIST_ACTION:
+                        return buildListOptions(DECK_MODEL);
+                }
+            default:
+                return null;
+        }
+    }
+
+    public static Options buildAddOptions() {
+        Options options = new Options();
+
+        Option questionOption = buildMultipleTokenOption("q", "question", true, "card question", true);
+        options.addOption(questionOption);
+
+        Option answerOption = buildMultipleTokenOption("a", "answer", true, "card answer", true);
+        options.addOption(answerOption);
+
+        return options;
+    }
+
+    public static Options buildTagOptions() {
+        Options options = new Options();
+        options.addOptionGroup(buildCardSelectOption());
+
+        Option tag = buildMultipleTokenOption("t", "tag", true, "tag name", true);
+        options.addOption(tag);
+
+        return options;
+    }
+
+    public static Options buildViewOptions() {
+        Options options = new Options();
+        options.addOptionGroup(buildCardSelectOption());
+
+        return options;
+    }
+
+    public static Options buildDeleteOptions(String model) {
+        Options options = new Options();
+        switch (model) {
```

Comment:
I think this is more confusing than having the Options being built in each XYZKeywordParser -- since our command looks like `<keyword> <action> flags`, once we parse the action we shouldn't need to look at the keyword anymore (so a more hierachial design).

If you wanted to go by the OOP route, perhaps something like this would be neater:
- Since `OptionsBuilder` does not really have a state (i.e., `model` and `action` are only used once, then a new `OptionsBuilder` is created the next time), maybe you could just do `static buildOptions(String keyword, String action)`
- Have an abstract `KeywordOptionBuilder`, which `XYZKeywordOptionBuilder` extends
- Move all the option building (e.g., `buildDeleteOptions`) to `XYZKeywordOptionBuilder`. `XYZKeywordOptionBuilder` will have a switch-case on how to handle each **action**
- `KeywordOptionBuilder.buildOptions` has a switch-case to handle each **keyword** and call the appropriate `XYZKeywordOptionBuilder.buildOption()`
- Functions that are used by multiple `XYZKeywordOptionBuilder` can be protected static functions in the abstract `KeywordOptionBuilder` (e.g., `buildCardSelectOption`)

This design is parallel to how `KeywordParser` and `XYZKeywordParser` are currently designed.

Reply:
Nevertheless, I think this is a good step in organizing how we're handling `Options` since that is getting quite messy. 👍